### PR TITLE
Location Hierarchy endpoint lineage ids enhancement and refactor ✨

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Resolves https://github.com/onaio/fhir-gateway-extension/issues/112
Resolves https://github.com/onaio/fhir-gateway-extension/issues/113

**Engineer Checklist**

- [x] I have written **Unit tests** for any new feature(s) and edge cases for
      bug fixes
- [x] I have added documentation for any new feature(s) and configuration
      option(s) on the `README.md`
- [x] I have run `mvn spotless:check` to check my code follows the project's
      style guide
- [x] I have run `mvn clean test jacoco:report` to confirm the coverage report
      was generated at `plugins/target/site/jacoco/index.html`
- [x] I ran `mvn clean package` right before creating this pull request.

### CODE REVIEW

File: exec/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.smartregister</groupId>
     <artifactId>opensrp-gateway-plugin</artifactId>
-    <version>2.2.6</version>
+    <version>2.3.0</version>
```
[the remark this answers is not in the document]
Comment (ndegwamartin) [reply exec/pom.xml:7]: Yeah they are, that's why I didn't increment

[the remark this answers is not in the document]
Comment (ndegwamartin) [reply exec/pom.xml:7]: Given all the changes we've made this far though, lets do a v3

**Review by lincmba — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]
